File: vtadm/vtadm.c
/*
 * Copyright 2010 Grigale Ltd. All right reserved.
 * Use is subject to license terms.
 */

#include <stdio.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <stdbool.h>
#include <stdlib.h>
#include <unistd.h>

#define	STMFSSD_CTL_DEV	"/dev/stmf_ssdctl"

typedef struct {
	int	fd;
} ctldev_t;

static ctldev_t ctldev;

bool
openctldev()
{
	ctldev.fd = open(STMFSSD_CTL_DEV, O_RDWR);
	if (ctldev.fd == -1) {
		/* No dice */
		return (false);
	}

	return (true);
}

void
closectldev()
{
	close(ctldev.fd);
}

int
parseopt(int argc, char **argv)
{
	if (argc == 1) {
		return (0);
	}
}

int
main(int argc, char **argv)
{
	printf("vtadm is not implemented yet!\n");

	parseopt(argc, argv);

	if (!openctldev()) {
		printf("Cannot open stmf_ssd control device\n");
		exit(EXIT_FAILURE);
	}

	closectldev();

	return (0);
}

File: vtadm/vtadm.py
#!/usr/bin/python2.6
#
# Copyright 2010 Grigale Ltd. All right reserved.
# Use is subject to license terms.
#
import fcntl
import optparse
import os

STMFSSDADMDEV = "/devices/pseudo/stmf_ssd@0:admin"

class VTAdmException(Exception):
    def __init__(self, msg=None):
        self.msg = msg

class StmfSsdAdmin():
    def __init__(self):
        try:
            self.ctldev = os.open(STMFSSDADMDEV, os.O_RDWR)
	except OSError:
            raise VTAdmException("Cannot open stmfssd control device")

    def status(self):
        fcntl.ioctl(self.ctldev, op, arg, mutate_flag)
	print "Status OK"


def parse():
    usage = "Usage: %prog verb [options]"
    version = "1"
    parser = optparse.OptionParser(usage, version=version)
    parser.add_option("status", dest="status")


def main():
    parse()
    ctldev = StmfssdAdmin()
    ctldev.status()


if __name__ == "__main__":
    try:
        main()
    except VTAdmException as e:
        print e.msg
